erc-7905.md

```diff
@@ -0,0 +1,226 @@
+ */
+interface IERC7905 {
+    /**
+     * @notice Reverts when a bridge with too low of a limit tries to call mint/burn
+     */
+    error IXERC20_NotHighEnoughLimits();
+
+    /**
+     * @notice Returns the max limit of a minter
+     *
+     * @param _bridge The bridge we are viewing the limits of
+     * @return _limit The limit the bridge has
+     */
+    function mintingMaxLimitOf(address _bridge) external view returns (uint256 _limit);
+
+    /**
+     * @notice Returns the max limit of a bridge
+     *
+     * @param _bridge the bridge we are viewing the limits of
+     * @return _limit The limit the bridge has
+     */
+    function burningMaxLimitOf(address _bridge) external view returns (uint256 _limit);
+
+    /**
+     * @notice Returns the current limit of a minter
+     *
+     * @param _bridge The bridge we are viewing the limits of
+     * @return _limit The limit the minter has
+     */
+    function mintingCurrentLimitOf(address _bridge) external view returns (uint256 _limit);
+
+    /**
+     * @notice Returns the current limit of a bridge
+     *
+     * @param _bridge the bridge we are viewing the limits of
+     * @return _limit The limit the bridge has
+     */
+    function burningCurrentLimitOf(address _bridge) external view returns (uint256 _limit);
+
+    /**
+     * @notice Mints tokens for a user
+     * @param _user The address of the user who needs tokens minted
+     * @param _amount The amount of tokens being minted
+     */
+    function mint(address _user, uint256 _amount) external;
+
+    /**
+     * @notice Burns tokens for a user
+     * @param _user The address of the user who needs tokens burned
+     * @param _amount The amount of tokens being burned
+     */
+    function burn(address _user, uint256 _amount) external;
+}
+```
+
+Implementations MUST additionally satisfy the following requirements:
+
+- `mint` MUST check that the caller's current available minting `limit` is greater than or equal to `_amount`
+- `mint` MUST increase the supply of the underlying ERC-20 token by `_amount` and reduce the current available `limit`
+- `burn` MUST check that the caller's current available burning `limit` is greater than or equal to `_amount`
+- `burn` MUST decrease the supply of the underlying ERC-20 token by `_amount` and reduce the current available `limit`
+
+## Rationale
+
+The proposed standard attempts to satisfy the following conditions for bridged tokens regardless of where and how they are bridged:
+
+1. Tokens received at the destination should be the canonical tokens.
+2. Bridges should not need to bootstrap liquidity for each new chain and asset.
+3. Cross-domain interactions involving tokens should be slippage-free, simplifying composability.
+4. Token issuers should own decision-making around which bridges to support and be able to parametrize risk for each. They should not be locked into only supporting a single bridge.
```

Comment:
This is describing the proposal as a whole, and not technical decisions made within it. This text therefore belongs in Motivation.